src/Datadog.Trace/ExtensionMethods/TagValueExtensions.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+
+namespace Datadog.Trace.ExtensionMethods
+{
+    internal static class TagValueExtensions
+    {
+        /// <summary>
+        /// Converts a <see cref="TagValue"/> into a <see cref="bool"/> by comparing it to commonly used values
+        /// such as "True", "yes", or "1". Case-insensitive. Defaults to <c>false</c> if string is not recognized.
+        /// </summary>
+        /// <param name="value">The TagValue to convert.</param>
+        /// <returns><c>true</c> if is one of the accepted values for <c>true</c>; <c>false</c> otherwise.</returns>
```

Comment:
I think the documentation for both this method and the `Datadog.Trace.ExtensionMethods.StringExtensions` need to be updated to reflect that they will return `null` if the input is not recognized as a string with a boolean value. And in this case also documenting that a double value will return true if 1, false otherwise.

Reply:
Ok, I'll change this documentation, and for `Datadog.Trace.ExtensionMethods.StringExtensions` the change should be in other PR isn't?

Reply:
Sure that change can go in a separate PR